vortexor/src/main.rs

```diff
@@ -1,23 +1,97 @@
 use {
-    clap::{value_t, value_t_or_exit},
+    clap::{crate_name, value_t, value_t_or_exit, values_t},
     crossbeam_channel::bounded,
+    log::*,
     solana_clap_utils::input_parsers::keypair_of,
     solana_core::banking_trace::BankingTracer,
-    solana_sdk::net::DEFAULT_TPU_COALESCE,
+    solana_net_utils::{bind_in_range_with_config, SocketConfig},
+    solana_sdk::{net::DEFAULT_TPU_COALESCE, signer::Signer},
     solana_streamer::streamer::StakedNodes,
     solana_vortexor::{
         cli::{app, DefaultArgs},
+        sender::{
+            PacketBatchSender, DEFAULT_BATCH_SIZE, DEFAULT_RECV_TIMEOUT,
+            DEFAULT_SENDER_THREADS_COUNT,
+        },
         vortexor::Vortexor,
     },
     std::{
+        collections::HashSet,
+        env,
         sync::{atomic::AtomicBool, Arc, RwLock},
+        thread::JoinHandle,
         time::Duration,
     },
 };
 
+#[cfg(unix)]
+fn redirect_stderr(filename: &str) {
+    use std::{fs::OpenOptions, os::unix::io::AsRawFd};
+    match OpenOptions::new().create(true).append(true).open(filename) {
+        Ok(file) => unsafe {
+            libc::dup2(file.as_raw_fd(), libc::STDERR_FILENO);
+        },
+        Err(err) => eprintln!("Unable to open {filename}: {err}"),
+    }
+}
+
+// Redirect stderr to a file with support for logrotate by sending a SIGUSR1 to the process.
+//
+// Upon success, future `log` macros and `eprintln!()` can be found in the specified log file.
+pub fn redirect_stderr_to_file(logfile: Option<String>) -> Option<JoinHandle<()>> {
```

Comment:
We seem to have identical function in the validator main. Do we really need to make a copy here?

Reply:
Moved to a utility lib to share code